forgerock-auth/src/main/java/org/forgerock/android/auth/SessionManager.java

```diff
@@ -52,17 +52,8 @@ public SessionManager(TokenManager tokenManager, SingleSignOnManager singleSignO
      */
     @WorkerThread
     public void refresh(FRListener<AccessToken> listener) {
-        getAccessToken(new FRListener<>() {
-            @Override
-            public void onSuccess(AccessToken result) {
-                tokenManager.refresh(result, listener);
-            }
-
-            @Override
-            public void onException(@NonNull Exception e) {
-                Listener.onException(listener, e);
-            }
-        });
+        AccessToken token = tokenManager.getAccessToken();
```

Comment:
if the accesstoken is null , it will throw the exception on the next line. it calling

``` String refreshToken = accessToken.getRefreshToken();
        if (refreshToken == null) {
            clear();
            Listener.onException(listener, new AuthenticationRequiredException("Refresh Token does not exists."));
            return;
        }```

is that what you are asking ?